File: lib/widget/widget_keys.dart
/// Cross-language contract for the home-widget SharedPreferences keys (TASK-111). These
/// must match the Kotlin `BgWidgetProvider` KEY_* constants exactly — a mismatch renders a
/// blank widget field with no compile-time error. Defined once and asserted equal to the
/// Kotlin side by a contract test.
library;

class WidgetKeys {
  static const String bgText = 'bg_text';
  static const String trend = 'bg_trend';
  static const String unit = 'bg_unit';
  static const String iob = 'iob_text';
  static const String range = 'bg_range';
  static const String cgmEpochMs = 'cgm_epoch_ms';

  /// The full set, for the contract test.
  static const Set<String> all = {bgText, trend, unit, iob, range, cgmEpochMs};
}

File: android/app/src/main/kotlin/com/bgdude/app/widget/WidgetKeys.kt
package com.bgdude.app.widget

/**
 * Single source of truth for the home-widget SharedPreferences keys shared with the Dart
 * side (TASK-111). Must match `lib/widget/widget_keys.dart` exactly — a mismatch renders a
 * blank widget field with no compile error. A contract test asserts [ALL] equals the Dart
 * set.
 */
object WidgetKeys {
    const val BG_TEXT = "bg_text"
    const val TREND = "bg_trend"
    const val UNIT = "bg_unit"
    const val IOB = "iob_text"
    const val RANGE = "bg_range"
    const val CGM_EPOCH_MS = "cgm_epoch_ms"

    val ALL: Set<String> = setOf(BG_TEXT, TREND, UNIT, IOB, RANGE, CGM_EPOCH_MS)
}
